src/Explorer/Tabs/QueryTab/QueryTabComponent.tsx

```diff
@@ -419,6 +434,16 @@ export default class QueryTabComponent extends React.Component<IQueryTabComponen
       });
     }
 
+    buttons.push({
+      iconSrc: DownloadQueryIcon,
+      iconAlt: "Download Query",
+      keyboardAction: KeyboardAction.DOWNLOAD_ITEM,
+      onCommandClick: this.onDownloadQueryClick,
+      commandButtonLabel: "Download Query",
+      ariaLabel: "Download Query",
+      hasPopup: false,
```

Comment:
Can we disable the button if query window is empty ?

Reply:
Yep, that should be possible.

Reply:
Done. What I actually did was tie the Download button in to the existing Save button state. That button wasn't disabled when the query was empty though, so I went ahead and added that logic since it seems appropriate to me. So now **both** the Save Query and Download Query buttons are disabled when the query editor is completely empty.